### PR TITLE
feat(badge): delete `outline` props and add `badgeType` props

### PR DESCRIPTION
**Pull request by yuda110**
Signed-off-by: Dahyun Yu <yuda@megazone.com>

### To Reviewers
- [ ] Skipping review is not a problem (```style```, ```chore``` ONLY)
- [ ] Not that difficult

### Type of Change
- [ ] New feature
- [ ] Bug fixes
- [x] Feature improvement
  - [x] These changes require a usage change
- [ ] Others (performance improvement, refactoring, CI/CD, etc.)

### Checklist
- [x] Updated Storybook documents for component
- [ ] Wrote test codes for composable, utils and etc.
- [ ] Tested with console(if usages are changed) 

### Description
- delete `outline` props
- add `badgeType` props

- before
  스크린샷 2023-02-13 오후 1 10 54

- after
  스크린샷 2023-02-13 오후 1 42 29

### CODE REVIEW

File: src/utils/helpers.ts [hunk 1]
```diff
@@ -41,8 +41,15 @@ export const isNotEmpty = (value): boolean => {
...
-    const color = get(colors, col);
+    let color = get(colors, col);
     if (color) return color;
+    // coral600 -> coral[600]
+    const colNum = col.match(/\d{3}/)?.[0];
+    if (colNum) {
+        const colStr = col.match(/[a-z]+/)?.[0];
+        color = get(colors, `${colStr}[${colNum}]`);
+        if (color) return color;
+    }
```
Comment (yuda110) [inline src/utils/helpers.ts:52]: I'm not sure if this is the right way. 😅

File: src/data-display/badge/PBadge.stories.mdx [hunk 2]
```diff
@@ -15,11 +15,12 @@ export const Template = (args,{argTypes})=>({
...
     template: `
         <div style="display:flex; align-items:center; justify-content:center; height:100px;">
-            <p-badge :styleType="styleType"
-                    :text-color="textColor"
-                    :backgroundColor="backgroundColor"
-                    :outline="outline"
-                    :shape="shape"
+            <p-badge
+                :badge-type="badgeType"
+                :styleType="styleType"
```
Comment (sulmoJ) [inline src/data-display/badge/PBadge.stories.mdx]: It's really clear!

File: src/data-display/badge/PBadge.vue [hunk 3]
```diff
@@ -96,51 +104,22 @@ export default defineComponent<BadgeProps>({
...
-    @mixin badge-without-outline red100, theme('colors.red.100'), theme('colors.red.500');
+    @mixin subtle primary3, theme('colors.primary3'), theme('colors.violet.600');
+    @mixin subtle blue200, theme('colors.blue.200'), theme('colors.blue.600');
+    @mixin subtle blue300, theme('colors.blue.300'), theme('colors.blue.600');
+    @mixin subtle green200, theme('colors.green.200'), theme('colors.green.700');
+    @mixin subtle indigo100, theme('colors.indigo.100'), theme('colors.indigo.600');
+    @mixin subtle gray200, theme('colors.gray.100'), theme('colors.gray.700');
+    @mixin subtle gray200, theme('colors.gray.200'), theme('colors.gray.900');
+    @mixin subtle yellow200, theme('colors.yellow.200'), theme('colors.gray.900');
+    @mixin subtle red100, theme('colors.red.100'), theme('colors.red.500');
```
Comment (sulmoJ) [inline src/data-display/badge/PBadge.vue:134]: 👍 👍 👍

**Review by sulmoJ — APPROVED** [review]
LGTM!!

**Pull request merged**